File: distros/J/JavonetPerlSdk/lib/Javonet/Binaries/Python/javonet/core/generator/handler/shared/SharedClassNameHandler.py
import keyword

from javonet.core.generator.handler.AbstractGeneretatorHandler import AbstractGeneratorHandler


class SharedClassNameHandler(AbstractGeneratorHandler):
    def generate_command(self, analyzed_object, parent_command, handlers):
        return analyzed_object

    def generate_code(self, existing_string_builder, common_command, used_object, handlers):
        if keyword.iskeyword(used_object):
            used_object += used_object[-1]
        if "@" in used_object:
            used_object = used_object.replace("@", "")
        if "-" in used_object:
            used_object = used_object.replace("-", "_")
        if used_object[0].isdigit():
            pos = 0
            for i, x in enumerate(used_object):
                if x.isdigit():  # True if its a number
                    pos = i  # first letter position
                break

            new_str = used_object[pos:]
            used_object = new_str
        if used_object[0].islower():
            used_object.capitalize()
        if used_object.isidentifier():
            existing_string_builder.append(used_object)
        else:
            raise NameError("SharedMethodNameHandlerError: Given name is not valid: " + used_object)

File: distros/J/JavonetPerlSdk/lib/Javonet/Binaries/Nodejs/core/generator/handler/shared/SharedClassNameHandler.js
const AbstractGeneratorHandler = require("../AbstractGeneratorHandler");
const reserved = require("reserved-words");

class SharedClassNameHandler extends AbstractGeneratorHandler {

    generate_code(existing_string_builder, common_command, used_object, handlers) {

        used_object = used_object.substring(used_object.lastIndexOf(".") + 1);

        if (used_object.includes('#')) {
            used_object = used_object.replace("#", "");
        }

        if (used_object.includes('-')) {
            used_object = used_object.replace("-", "_");
        }

        if (this.is_numeric_char(used_object[0])) {
            let pos = 0;
            for (const [index, element] of used_object.entries()) {
                if (this.is_numeric_char(element)) {
                    pos = index
                }
                break
            }
            const new_str = used_object.slice(pos);
            used_object = new_str
        }

        if (used_object[0] === used_object[0].toUpperCase()) {
            used_object = used_object[0].toLowerCase() + used_object.slice(1)
        }

        if (!reserved.check(used_object)) {
            existing_string_builder.append(used_object)
        } else {
            throw ("SharedMethodNameHandlerError: Given name is not valid: " + used_object)
        }
    }

    is_numeric_char(c) {
        return /\d/.test(c);
    }
}

module.exports = SharedClassNameHandler